lib/components/primitive-components/Group/Group.ts

```diff
@@ -121,14 +113,20 @@ export class Group<Props extends z.ZodType<any, any, any> = typeof groupProps>
     const serverMode = this.props.autorouter?.serverMode ?? "job"
 
     const debug = Debug("tscircuit:core:autorouting")
+    debug("props", this.props.autorouter)
 
     const fetchWithDebug = (url: string, options: RequestInit) => {
       debug("fetching", url)
       return fetch(url, options)
     }
 
-    // Queue the autorouting request
-    this._queueAsyncEffect("make-http-autorouting-request", async () => {
+    // Wait for any existing autorouting to complete
+    if (Group.currentAutoroutingPromise) {
+      await Group.currentAutoroutingPromise
+    }
```

Comment:
i get the idea with this method, but unrelated subcircuits should be able to go in parallel